pkg/sql/parser/sql.y

```diff
@@ -7261,6 +7269,38 @@ simple_typename:
 | character_with_length
 | interval_type
 | postgres_oid
+| POINT error { return unimplementedWithIssueDetail(sqllex, 21286, "point") } // needed or else it generates a syntax error.
+| POLYGON error { return unimplementedWithIssueDetail(sqllex, 21286, "polygon") } // needed or else it generates a syntax error.
+
+geo_shape:
+  POINT { $$.val = geopb.Shape_Point }
+| LINESTRING { $$.val = geopb.Shape_LineString }
+| POLYGON { $$.val = geopb.Shape_Polygon }
+| GEOMETRYCOLLECTION { $$.val = geopb.Shape_GeometryCollection }
+| MULTIPOLYGON { $$.val = geopb.Shape_MultiPolygon }
+| MULTILINESTRING { $$.val = geopb.Shape_MultiLineString }
+| MULTIPOINT { $$.val = geopb.Shape_MultiPoint }
+| GEOMETRY { $$.val = geopb.Shape_Geometry }
```

Comment:
I'd expect there to be parse tests for every one of these cases, to prevent regressions